### PR TITLE
S3 listing of object versions and download options

### PR DESCRIPTION
**Pull request by bpasquer**
this is my take on the S3 listing and download of previous versions

### CODE REVIEW

File: lib/python/s3_ls_prefix.py [hunk 1]
```diff
@@ -69,21 +87,45 @@ def args():
                         type=str,
                         help="download listed objects to --output-dir")
 
+    parser.add_argument("-l", "--listversion",
+                        type=str,
+                        help="list object version --listversion")
+
+    parser.add_argument("-v", "--versionid", nargs=2 ,
```
Comment (lbesnard) [inline lib/python/s3_ls_prefix.py:94]: I believe nargs should be =1

File: lib/python/s3_ls_prefix.py [hunk 2]
```diff
@@ -7,18 +7,20 @@
...
   1) set up credentials with `aws configure` (binary coming from awscli package)
   2) download file
      https://github.com/aodn/cloud-deploy/blob/master/sample-config/aws/config_projectofficer
      copy and rename to ~/.aws/config
-
 Examples:
   ./s3_ls_prefix.py -p 'Defence_Technology_Agency-New_Zealand/Waverider_Buoys_C-20200615T000000Z/'
   ./s3_ls_prefix.py -b imos-test-data -p 'Defence_Technology_Agency'
   ./s3_ls_prefix.py -p IMOS/ACORN/gridded_1h-avg-current-map_QC/NWA/2021/07/ -o /tmp/nwa_dm
+   ./s3_ls_prefix.py -l CSIRO/Climatology/Ocean_Acidification/OA_Reconstruction.nc
```
Comment (lbesnard) [inline lib/python/s3_ls_prefix.py:22]: indent

File: lib/python/s3_ls_prefix.py [hunk 3]
```diff
@@ -47,6 +49,22 @@ def _download_s3_obj_flat(bucket, s3_obj_ls, output_dir):
...
+      raise Exception("boto3 client error in list_all_objects_version function: " + e.__str__())
+   except Exception as e:
+      raise Exception("Unexpected error in list_all_objects_version function of s3 helper: " + e.__str__())
+
+   return result
+
+def _download_file_version(bucket, prefix, versionid):
+    s3 = boto3.client('s3')
+    filename = os.path.basename(prefix)
+    s3.download_file(bucket, prefix, filename, ExtraArgs={'VersionId': versionid})
```
Comment (lbesnard) [inline lib/python/s3_ls_prefix.py:66]: the -o option for output folder is not used anymore

**Review by lbesnard — CHANGES_REQUESTED** [review]

**Comment by bpasquer** [comment]
Instead of using this python script, it's been requested that awcli is installed in prod, as it is providing a range of functionalities to access S3 version information.
see comments https://github.com/aodn/issues/issues/1321

Closing

**Pull request closed without merging**